test/sequential/bootstrap.checkActivatePool.test.ts

```diff
@@ -0,0 +1,209 @@
+  }
+
+  bootstrapPhase = await api.query.bootstrap.phase();
+  expect(bootstrapPhase.toString()).toEqual("BeforeStart");
+});
+
+test("bootstrap - Check that we can not create a pool for the bootstrap token after bootstrap was declared", async () => {
+  const api = getApi();
+  await setupApi();
+
+  bootstrapPhase = await api.query.bootstrap.phase();
+  if (bootstrapPhase.toString() === "Finished") {
+    const bootstrapFinalize = await finalizeBootstrap(sudo);
+    eventResponse = getEventResultFromMangataTx(bootstrapFinalize);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  }
+
+  bootstrapPhase = await api.query.bootstrap.phase();
+  expect(bootstrapPhase.toString()).toEqual("BeforeStart");
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(bootstrapCurrency, testUser1), // transferAll test
+    Assets.mintToken(bootstrapCurrency, sudo), // transferAll test
+    Assets.mintNative(testUser1)
+  );
+
+  await sudo.mint(bootstrapCurrency, testUser1, toBN("1", 20));
+  const sudoBootstrap = await scheduleBootstrap(
+    sudo,
+    MGA_ASSET_ID,
+    bootstrapCurrency,
+    waitingPeriod,
+    bootstrapPeriod,
+    whitelistPeriod
+  );
+  eventResponse = getEventResultFromMangataTx(sudoBootstrap);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+  await checkingPool();
+
+  await waitForBootstrapStatus("Whitelist", waitingPeriod);
+
+  await checkingPool();
+
+  await waitForBootstrapStatus("Public", waitingPeriod);
+
+  await checkingPool();
+
+  // new token must participate in provision as first
+  const provisionPublicBootstrapCurrency = await provisionBootstrap(
+    testUser1,
+    bootstrapCurrency,
+    bootstrapAmount
+  );
+  eventResponse = getEventResultFromMangataTx(provisionPublicBootstrapCurrency);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+  // we need to add MGA token in the provision for creating a pool
+  const provisionPublicMGA = await provisionBootstrap(
+    testUser1,
+    MGA_ASSET_ID,
+    bootstrapAmount
+  );
+  eventResponse = getEventResultFromMangataTx(provisionPublicMGA);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+  await waitForBootstrapStatus("Finished", bootstrapPeriod);
+
+  await checkingPool();
+
+  const provisionFinished = await provisionBootstrap(
+    testUser1,
+    bootstrapCurrency,
+    bootstrapAmount
+  );
+  eventResponse = getEventResultFromMangataTx(provisionFinished);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+  expect(eventResponse.data).toContain("Unauthorized");
+
+  // Check existing pool
+  bootstrapPool = await api.query.xyk.pools([MGA_ASSET_ID, bootstrapCurrency]);
+  expect(bootstrapPool[0]).bnEqual(bootstrapAmount);
+  expect(bootstrapPool[1]).bnEqual(bootstrapAmount);
+  const bootstrapPoolBalance = new BN(
+    bootstrapPool[0].add(bootstrapPool[1]) / 2
+  );
+
+  // need claim liquidity token before finalizing
```

Comment:
can you move all this to an afterEach() function? 
The test has finished, it's just about leaving the environment in a good state.